### PR TITLE
Fixed #25109 -- Added a system check for invalid entries in MIGRATION_MODULES.

### PR DESCRIPTION
**Pull request by berkerpeksag**
https://code.djangoproject.com/ticket/25109

### CODE REVIEW

File: django/core/checks/database.py [hunk 1]
```diff
@@ -9,3 +11,23 @@ def check_database_backends(*args, **kwargs):
...
+
+
+@register(Tags.database)
+def check_migration_modules(*args, **kwargs):
+    from django.conf import settings
+    migration_modules = getattr(settings, 'MIGRATION_MODULES', {})
+    warnings = []
+    for app_label, module in migration_modules.items():
+        if module is None:
+            continue
```
Comment (MarkusH) [inline django/core/checks/database.py]: Why do you continue here? `app_label` might still be invalid.

File: django/core/checks/database.py [hunk 2]
```diff
@@ -9,3 +11,23 @@ def check_database_backends(*args, **kwargs):
...
         issues.extend(conn.validation.check(**kwargs))
     return issues
+
+
+@register(Tags.database)
+def check_migration_modules(*args, **kwargs):
+    from django.conf import settings
+    migration_modules = getattr(settings, 'MIGRATION_MODULES', {})
+    warnings = []
+    for app_label, module in migration_modules.items():
```
Comment (MarkusH) [inline django/core/checks/database.py]: the keys are not necessarily app labels but might as well be paths to app configs. Maybe replace `app_label` with `app`?

File: django/core/checks/database.py [hunk 3]
```diff
@@ -9,3 +11,23 @@ def check_database_backends(*args, **kwargs):
...
+        if module is None:
+            continue
+        try:
+            importlib.import_module(module)
```
Comment (MarkusH) [inline django/core/checks/database.py]: I'd rather check for the app being in `INSTALLED_APPS` than if it's importable.

**Comment by charettes** [comment]
@MarkusH, @berkerpeksag what do you think of the alternative I suggested [here](https://code.djangoproject.com/ticket/25109#comment:5)?

**Comment by timgraham** [comment]
The alternative seems better to me. Let's reopen this if it turns out not to be.

**Pull request closed without merging**